dotnet/src/SemanticKernel.Core/KernelExtensions.cs

```diff
@@ -364,108 +363,68 @@ public static ISKPlugin ImportPluginFromObject(this Kernel kernel, object target
-        Verify.NotNull(kernel);
-
-        return kernel.RunAsync(new ContextVariables(), pipeline);
-    }
-
-    /// <summary>
-    /// Run a pipeline composed of synchronous and asynchronous functions.
+    /// Run a single synchronous or asynchronous <see cref="KernelFunction"/>.
     /// </summary>
     /// <param name="kernel">The kernel.</param>
+    /// <param name="function">A Semantic Kernel function to run</param>
     /// <param name="input">Input to process</param>
-    /// <param name="pipeline">List of functions</param>
-    /// <returns>Result of the function composition</returns>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>Result of the function</returns>
     public static Task<FunctionResult> RunAsync(
         this Kernel kernel,
+        KernelFunction function,
         string input,
-        params KernelFunction[] pipeline)
+        CancellationToken cancellationToken = default)
     {
         Verify.NotNull(kernel);
+        Verify.NotNull(function);
 
-        return kernel.RunAsync(new ContextVariables(input), pipeline);
-    }
-
-    /// <summary>
-    /// Run a pipeline composed of synchronous and asynchronous functions.
-    /// </summary>
-    /// <param name="kernel">The kernel.</param>
-    /// <param name="variables">Input to process</param>
-    /// <param name="pipeline">List of functions</param>
-    /// <returns>Result of the function composition</returns>
-    public static Task<FunctionResult> RunAsync(
-        this Kernel kernel,
-        ContextVariables variables,
-        params KernelFunction[] pipeline)
-    {
-        Verify.NotNull(kernel);
+        var contextVariables = new ContextVariables();
+        contextVariables.Update(input);
 
-        return kernel.RunAsync(variables, CancellationToken.None, pipeline);
+        return kernel.RunAsync(function, contextVariables, cancellationToken);
     }
 
     /// <summary>
-    /// Run a pipeline composed of synchronous and asynchronous functions.
+    /// Run a single synchronous or asynchronous <see cref="KernelFunction"/>.
     /// </summary>
     /// <param name="kernel">The kernel.</param>
+    /// <param name="function">A Semantic Kernel function to run</param>
+    /// <param name="variables">Input to process</param>
     /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
-    /// <param name="pipeline">List of functions</param>
-    /// <returns>Result of the function composition</returns>
+    /// <returns>Result of the function</returns>
     public static Task<FunctionResult> RunAsync(
         this Kernel kernel,
-        CancellationToken cancellationToken,
-        params KernelFunction[] pipeline)
+        KernelFunction function,
+        ContextVariables variables,
```

Comment:
This will allow all to use 
RunAsync(function)
RunAsync(function, variables)
RunAsync(function, variables, cancellationToken)

```suggestion
        ContextVariables? variables = null,
```

Reply:
The code already supports these

```
RunAsync(function)
RunAsync(function, input)
RunAsync(function, input, cancellationToken)
RunAsync(function, variables)
RunAsync(function, variables, cancellationToken)
```